sporkfish/zobrist_hasher.py

```diff
@@ -40,82 +185,93 @@ def __init__(self, board: Board) -> None:
+        castling_rights = ZobristHasher._parse_castling_rights(board)
 
-    for i, castling_right in enumerate(castling_rights):
-        if castling_right:
-            board_hash ^= castling_hashes[i]  # type: ignore
+        zobrist_hash = _full_zobrist_hash(  # type: ignore
+            squares, colored_piece_types, board.turn, ep_file, castling_rights
+        )
+        return ZobristStateInfo(zobrist_hash, ep_file, castling_rights)
 
-    return int(board_hash)
+    def incremental_zobrist_hash(
+        self,
+        board: Board,
+        move: chess.Move,
+        prev_state: ZobristStateInfo,
+        previous_from_square_piece: chess.Piece,
+        captured_piece: Optional[chess.Piece],
+    ) -> ZobristStateInfo:
+        """
+        Compute the Zobrist hash value incrementally after a move.
 
+        :param board: The chess board.
+        :type board: Board
+        :param move: The move being made.
+        :type move: chess.Move
+        :param prev_state: The previous Zobrist state information.
+        :type prev_state: ZobristStateInfo
+        :param previous_from_square_piece: The piece from the originating square of move.
+        :type previous_from_square_piece: chess.Piece
+        :param captured_piece: The piece captured, if any, defaults to None
+        :type captured_piece: Optional[chess.Piece]
 
-class ZobristHasher:
-    def __init__(self) -> None:
+        :return: An object containing the updated Zobrist hash value and other board state information.
+        :rtype: ZobristStateInfo
         """
-        Initialize the Zobrist Hasher object, used to hash board positions.
-        This allows caching via the transposition table, so we don't have to evaluate positions twice.
-
-        Attributes:
-        - _piece_hashes (numpy.ndarray): Zobrist hash values for each piece on each square.
-        - _turn_hash (np.uint64): Zobrist hash value for the side to move.
-        - _en_passant_hashes (numpy.ndarray): Zobrist hash values for en passant squares.
-        - _castling_hashes (numpy.ndarray): Zobrist hash values for castling rights.
-
-        Methods:
-        - __init__(): Initialize the Zobrist Hasher object.
-        - hash(board) -> int: Hashes the given chess board using Zobrist hashing.
-        """
-        # Sporkfish's lucky number
-        np.random.seed(10101010)
+        from_color_piece_type = hash(previous_from_square_piece)
 
-        self._piece_hashes = np.random.randint(0, 2**64, size=(64, 12), dtype=np.uint64)
-        self._turn_hash = np.random.randint(0, 2**64, dtype=np.uint64)
+        # XOR out the previous from square piece
+        squares_list = [move.from_square]
+        colored_piece_types_list = [from_color_piece_type]
 
-        self._en_passant_hashes = np.random.randint(0, 2**64, size=8, dtype=np.uint64)
-        self._castling_hashes = np.random.randint(0, 2**64, size=4, dtype=np.uint64)
+        # XOR in the to square piece, piece type depending on if promoted or not
```

Comment:
What's the difference between "XOR in" vs "XOR out"?

Very minor, but would prefer if this line of comment is moved down a line?

Reply:
Both are same in terms of mathjematical operation but easier for me to think about. In means its the singularly in the hash whereas out is if theres two copies in the hash (i.e. it cancels out).